### PR TITLE
State machines and regions

### PR DESCRIPTION
**Pull request by amolenaar**
### PR Type
What kind of change does this PR introduce?

- [ ] Bug fix
- [x] Feature
- [ ] Chore (refactoring, formatting, local variables, other cleanup)
- [ ] Documentation content changes

### What is the current behavior?

State machines are only created implicitly to hold states.

Issue Number: #1654 

### What is the new behavior?

- [x] State machines can be created from the toolbox
- [x] (pseudo)states can be moved between state machines
- [x] Region support in Statemachine (state / pseudostate / transition)
- [x] Region support in (composite) State
- [x] property page for regions (create / remove) - how to deal with elements in the region? 
- [x] create state / pseudostate / transition from toolbox places it in the right region

### Other

This PR contains a changes where stereotype watch and compartment code is directly imported from the stereotype module.

### CODE REVIEW

File: gaphor/UML/states/propertypages.py [hunk 1]
```diff
@@ -11,6 +14,9 @@
     handler_blocking,
     new_resource_builder,
 )
+from gaphor.UML.states.state import StateItem
+
+from .statemachine import StateMachineItem
```
Comment (danyeaw) [inline gaphor/UML/states/propertypages.py:19]: To stay consistent with the other import statements, should we use absolute imports here?

**Review by danyeaw — COMMENTED** [review]
I got about 1/2 way through this, and it looks great so far!

File: gaphor/UML/states/propertypages.py (hunk 1 above)
Comment (amolenaar) [reply gaphor/UML/states/propertypages.py:19]: Yes it should. I suppose VSCode tried to help me by importing this item

**Review by danyeaw — APPROVED** [review]
Wow! Nice improvement to the state machine functionality!

**Pull request merged**